pkg/controller/windowsmachineconfig/nodeconfig/nodeconfig.go

```diff
@@ -48,6 +51,53 @@ func (nc *nodeConfig) Configure() error {
 	if err := nc.handleCSRs(); err != nil {
 		return errors.Wrap(err, "handling CSR for the given node failed")
 	}
+
+	var instanceID string
+	// As the CSR has been approved get the Kubernetes node object associated
+	// the Windows VM created
+	if nc.Windows.GetCredentials() != nil && len(nc.Windows.GetCredentials().GetInstanceId()) > 0 {
+		instanceID = nc.Windows.GetCredentials().GetInstanceId()
+	}
+
```

Comment:
So what happens if you are unable to get an `instanceID`? You need a separate error check for that.

Reply:
You need to handle all the error cases. If the credentials are nil this will fall through. Please rewrite your check above to return an error if credentials are nil or instanceID is blank.